modules/unpickler/src/main/scala/ch/epfl/scala/debugadapter/internal/stacktrace/Scala3Unpickler.scala

```diff
@@ -237,24 +237,46 @@ class Scala3Unpickler(
     if method.declaringType.isObject && !method.isExtensionMethod then obj.headOption else cls.headOption
 
   private def findSymbolsRecursively(owner: DeclaringSymbol, encodedName: String): Seq[ClassSymbol] =
-    owner.declarations
-      .collect { case sym: ClassSymbol => sym }
-      .flatMap { sym =>
-        val encodedSymName = NameTransformer.encode(sym.name.toString)
-        val Symbol = s"${Regex.quote(encodedSymName)}\\$$?(.*)".r
-        encodedName match
-          case Symbol(remaining) =>
-            if remaining.isEmpty then Some(sym)
-            else findSymbolsRecursively(sym, remaining)
-          case _ => None
-      }
+    val pattern = """^([^$]+)[$](\d+)[$]?(.*)$""".r
```

Comment:
A few comments about the regex:
- I would rather use `\` to escape the `$`
- The final `$` is useless because `.*` matches everything up to the end
- The entire final `\$.*` is optional, not just the `\$` separator

```suggestion
    val pattern = """^([^$]+)\$(\d+)(\$.*)?$""".r
```

`remaining` would then start with `$` though. You can strip it with `remaining.stripPrefix("$")`